### ISSUE TITLE
Enhance election lifecycle clarity

### ISSUE THREAD

**Issue by bedeho** [Joystream/pioneer#4006]
# Background

Currently, the election works like this

```
Announcing Period: 14 days
Voting Period: 3 days
Reveal Period: 3 days
Idle Period: 1 days
Total: 21 days = 3 weeks
```

However, when you land on the election screen, you jus see the current period and how much time is left. This makes it hard for people to understand the full cycle, in terms of duration and structure, and this deters participation from both prospective candidates and voters.

# Proposal

Add some sort of visual representation to this screen which solves this.

# Solution

![image](https://user-images.githubusercontent.com/6571453/230619986-380d8b69-2681-46dd-afdc-15c771a09e2a.png)

**Comment by 0x2bc** [comment]
We should re-check the length of each period, since it appears to be different on the blockchain when compared to what was stated in the mainnet parameters document. For more details, please refer to the following discussion https://discord.com/channels/811216481340751934/813361923172335648/1060165808875114566

**Comment by traumschule** [comment]
Actual lengths can be gathered through the api via [hook](https://github.com/Joystream/pioneer/tree/dev/packages/ui/src/council/hooks) `useCouncilConstants.ts`.
For future refactoring: number of hooks may be reduced for better maintainability.

**Comment by mkbeefcake** [comment]
![image](https://user-images.githubusercontent.com/69617937/211775163-8ba50ca7-8e62-4a6b-80f6-3f875bad6d15.png)
I have added Election period section for lifecycle. please check this screenshot

**Comment by toiletgranny** [comment]
Alright, so yesterday @dmtrjsg asked me to weigh in here and come up with some ideas, as a designer. And so, guys, my recommendations for addressing this issue are ready to be reviewed! Find out all about it from:

- [This Loom video update](https://www.loom.com/share/277b80ce1e1c4c43a1431ee705fd18b7) (~5 min at 1.5x speed)
- [This Figma prototype](https://www.figma.com/proto/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-design?page-id=8328%3A230599&node-id=11634%3A438508&viewport=-800%2C-1987%2C0.06&scaling=scale-down&starting-point-node-id=11634%3A438508&show-proto-sidebar=1).

**Comment by dmtrjsg** [comment]
There's revealing period as well. But we concluded version 1 is better 👍 

Also re Idle period - right now there is no election page for the idle period and the app goes to Past votes, as the election is actually over.. So we suggest some sort of a placholder screen for this phase that is not included into the timeline..

**Comment by bedeho** [comment]
Fantastic work, I think both are great, I guess option 1 is less work, so I think that may be the best one to go with.

**Comment by toiletgranny** [comment]
Thanks! @dmtrjsg Is there any option for me to actually see what the page looks like in the idle period? What would be your idea of a placeholder screen for this view? Would that make sense, for the idle state, to show who actually won the election? Or is this information surfaced during the revealing period?

**Comment by traumschule** [comment]
>  right now there is no election page for the idle period and the app goes to Past votes

This behavior is a bit clumsy because it also happens during loading and then stays there.

**Comment by traumschule** [comment]
> * [This Loom video update](https://www.loom.com/share/277b80ce1e1c4c43a1431ee705fd18b7) (~5 min at 1.5x speed)
> * [This Figma prototype](https://www.figma.com/proto/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-design?page-id=8328%3A230599&node-id=11634%3A438508&viewport=-800%2C-1987%2C0.06&scaling=scale-down&starting-point-node-id=11634%3A438508&show-proto-sidebar=1).

Thanks a lot for the fast response, excellent as always!
What do you all think about both? :)

At 5:00 you mention the possible lack of clarity to the user that the bar is clickable. We could go with A as default (always visible) and open B as sidebar when user clicks on an election stage (and scroll to / highlight the according section).
To increase clarity there could be decent colors or a fine vertical black bar between them as separator.

**Comment by dmtrjsg** [comment]
@traumschule thank you for the feedback, but let's allow @toiletgranny to drive this forward to completion 😇

**Comment by dmtrjsg** [comment]
> Would that make sense, for the idle state, to show who actually won the election? Or is this information surfaced during the revealing period?

Previous election is now complete. View past elections / View current council links and a short message along the lines of  Announcing Period starts in ..

**Comment by toiletgranny** [comment]
Alright, this is ready to be reviewed and implemented. 

https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-design?node-id=11664%3A445586&t=btpGbvJr9qG6E7JF-1

@dmtrjsg For the Idle stage, I went for displaying voting results of the — technically — still _current_ round of elections while keeping the page layout the same. Rather than referring to it as, literally, _the idle state_, I propose to count down time till the next round of elections. Hope you like this idea more! Also, for the CTAs in the top right corner, my suggestion is to offer the option to restore votes, hoping this makes sense technically. 

To whoever will be implementing this, I highly recommend going over _all_ implementation notes I left in Figma (in the form of Figma comments) _before_ starting the actual coding. To enable comments visibility in Figma use the `SHIFT + C` keyboard shortcut. 
![CleanShot 2023-01-12 at 15 04 16](https://user-images.githubusercontent.com/35307309/212087241-2af21546-b36c-4a45-9870-dfd982340660.png)

**Comment by dmtrjsg** [comment]
@toiletgranny this information is already contained in the past votes (Second tab),  just checking that you are suggesting to show the actual results on that page as well, as opposed to the link?

I guess that would also work, @traumschule would you agree that it would be the most common use case for the idle election stage?

**Comment by toiletgranny** [comment]
@dmtrjsg That's correct, that's my recommendation. I reckon presenting results of the last (technically: current) round of elections, along with a prominent countdown till the next round, is a better use of the space. Let me know if you'd like me to change that.

**Comment by thesan** [comment]
@chrlschwb IMO you might want to raise that `SP:2`

**Comment by mkbeefcake** [comment]
@thesan I am working on this UI and checked the Anouncing days : 9 days, Voting and Revealing Dats : 3 days and idlePeriod is not existing. I hope to confirm this with you and modification for UI

**Comment by chrlschwb** [comment]
@mkbeefcake the exact time of each period should be a variable. They can change with runtime upgrades.

**Comment by msmadeline** [comment]
@dmtrjsg @chrlschwb Hello! I updated the RWD for election page with the enhancements from this isue. I propose to you a sollution that allows user to have the same experience on the mobile as on desktop. I attached the screenshot of both sollutions below. version 1 would not allow user to get all the information on the mobile devices as they can on desktop - that's why I propose verion 2 which adds accordeon components so the user can check the status of the election process and has the same experience.
![image](https://github.com/Joystream/pioneer/assets/139866824/06a6ae73-8379-464f-a709-7a3f81e07783)
![image](https://github.com/Joystream/pioneer/assets/139866824/d9e42b61-efe7-4dde-b2e4-cb9be201ae1e)
